src/coreclr/jit/emitarm64.cpp

```diff
@@ -8908,6 +8908,7 @@ void emitter::emitIns_Call(EmitCallType          callType,
 
     dispIns(id);
     appendToCurIG(id);
+    emitLastMemBarrier = nullptr; // Cannot optimize away future memory barriers
```

Comment:
I think this entire peephole should be switched to use the new backwards traversal peephole mechanism, which I expect would have quite significant TP improvements as we currently have code in very hot `appendToCurIG` to handle this peephole. I've opened #91825 for that. However, given that this PR should be backported I've kept the fix here surgical.